Neos.ContentRepository.Core/Classes/Feature/Common/NodeReferencingInternals.php

```diff
@@ -0,0 +1,43 @@
+<?php
+namespace Neos\ContentRepository\Core\Feature\Common;
+
+use Neos\ContentRepository\Core\Feature\NodeReferencing\Dto\NodeReferenceNameToEmpty;
+use Neos\ContentRepository\Core\Feature\NodeReferencing\Dto\NodeReferencesToWrite;
+use Neos\ContentRepository\Core\Feature\NodeReferencing\Dto\SerializedNodeReference;
+use Neos\ContentRepository\Core\Feature\NodeReferencing\Dto\SerializedNodeReferences;
+use Neos\ContentRepository\Core\Infrastructure\Property\PropertyConverter;
+use Neos\ContentRepository\Core\NodeType\NodeType;
+use Neos\ContentRepository\Core\NodeType\NodeTypeName;
+
+/**
+ *
+ */
+trait NodeReferencingInternals
+{
+    abstract protected function getPropertyConverter(): PropertyConverter;
+    abstract protected function requireNodeType(NodeTypeName $nodeTypeName): NodeType;
+
+    private function mapNodeReferencesToSerializedNodeReferences(NodeReferencesToWrite $references, NodeTypeName $nodeTypeName): SerializedNodeReferences
+    {
+        $serializedReferences = [];
+        foreach ($references->getIterator() as $reference) {
```

Comment:
```suggestion
        foreach ($references as $reference) {
```

isnt it part of the `IteratorAggregate` thing that one doesnt have to do it?

Reply:
right, right, I messed with this a bit, there are probably more artifacts like this in there.